chains/solana/contracts/programs/ccip-router/src/instructions/v1/offramp.rs

```diff
@@ -562,8 +561,13 @@ fn get_token_accounts_for<'a>(
 // should_execute_messaging checks if:
 // 1. There is at least one account used for messaging (the first subset of accounts). This is because the first account is the program id to do the CPI
 // 2. AND the logic_receiver has a value different than zeros
-fn should_execute_messaging(logic_receiver: &Pubkey, remaining_accounts_empty: bool) -> bool {
-    !remaining_accounts_empty && *logic_receiver != Pubkey::default()
+fn should_execute_messaging<'a>(
+    remaining_accounts: &'a [AccountInfo<'a>],
+    token_indices: &[u8],
+) -> bool {
+    // The first entry in the accounts corresponds to a token, which means there is no logic receiver
+    let only_tokens = token_indices.first().map(|i| *i == 0).unwrap_or_default();
```

Comment:
does the `unwrap_or_default` defaults to false?

Reply:
Yep it does! So if there are no token indices, this would be false.